src/hotspot/share/services/mallocTracker.hpp

```diff
@@ -291,8 +291,11 @@ class MallocTracker : AllStatic {
   static void* record_malloc(void* malloc_base, size_t size, MEMFLAGS flags,
     const NativeCallStack& stack);
 
-  // Record free on specified memory block
-  static void* record_free(void* memblock);
+  // Record free on specified memory block.
+  // Returns the outer pointer to this block.
+  static void* record_free_block(void* memblock);
+  // Record a free.
+  static void record_free(FreePackage free_package);
```

Comment:
Can you expand on the comment a bit?

`Given a block returned by os::malloc() or os::realloc(), de-account block from NMT, mark its header as dead and return pointer to header.`.

`Given the free info from a block, de-account block from NMT.`.